source/core/StringUtilities.cpp

```diff
@@ -1075,13 +1075,13 @@ void SubString::TrimQuotedString(TokenTraits tt)
 IntIndex SubString::FindFirstMatch(SubString* searchString, IntIndex offset, Boolean ignoreCase)
 {
     IntIndex searchStringLength = searchString->Length();
-    if (searchStringLength > Length())
+    if (Length() == 0 || searchStringLength > Length())
         return -1;
     if (offset < 0) {
         offset = 0;
     }
     const Utf8Char* pStart = _begin + offset;
-    const Utf8Char* pEnd = _end - searchStringLength;
+    const Utf8Char* pEnd = _end - searchStringLength+1;
```

Comment:
I am assuming the intent here is not "_end - (searchStringLength+1)". With associativity of + and - operators, this code is misleading to read with no spacing around +.

Reply:
Good catch. Looks like SearchSplitString and SearchAndReplaceString are affected by this change and may exhibit the same problem. I would recommend adding a couple of test cases that shows the problem without this fix to test-it\ViaTests\SearchSplitString.via. Carlos or I will be able to create a .via file with tests for SearchSplitString which may also be checked in.

Reply:
Fixed spacing and added test case to SearchSplitString.via.
I do not think a "couple" of test cases are necessary.